components/engine_traits/src/engine.rs

```diff
@@ -282,6 +301,11 @@ pub trait TabletFactory<EK>: TabletAccessor<EK> + Send + Sync {
         unimplemented!();
     }
 
+    /// Unregister the tablet if it is not used
+    fn unregister_tablet(&self, _region_id: u64, _suffix: u64) {
```

Comment:
When will it be used?

Reply:
After the derived peer load the new cloned tablet in factory, it is used to unregister the old one.

Reply:
Why not only keeping one tablet for one region?

Reply:
This is exactly the reason that I need it.

Reply:
Why not index the tablet by region ID only?

Reply:
This is used to clear cache only.
After creating the checkpoint of the tablet, source peer opens the new cloned tablet (ex: old tablet -> 1_5, and new cloned one -> split_1_10) .  The source region opens the split_1_10, so there are two tablets reserved in the cache (1_5, split_1_10).  `unregister_tablet` is used to remove 1_5 from cache.

Reply:
"What I mean is when register 1_10, 1_5 should be removed automatically."
How this can be done, by traversing?

Reply:
By "index the tablet by region ID only".

Reply:
You mean change the signature of `fn open_tablet(id, suffix, options)` to `fn open_tablet(id, options)` ?

Reply:
No, by changing the type of registry from `Arc<Mutex<HashMap<(u64, u64), RocksEngine>>>` to `Arc<Mutex<HashMap<u64, RocksEngine>>>`

Reply:
Okay, I see.